src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import duke.tasklist.TaskList;
+
+import duke.tasks.Task;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.ObjectInputStream;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Storage class to handle file storage, creation and reading of saved file.
+ */
+public class Storage {
+
+    static final String FILE_PATH = "./data/dataFile";
+
+    TaskList taskList;
+
+    /**
+     * Constructor of the duke.Storage class.
+     *
+     * @param taskList TaskList object being used to maintain the list of tasks in the chatbot.
+     */
+    public Storage(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+
+    /**
+     * Saves the current history array to the file
```

Comment:
Just a small nit with the Javadocs:

```suggestion
     * Saves the current history array to the file.
```